src/components/Icon/Icon.astro

```diff
@@ -7,19 +7,21 @@ interface Props extends HTMLAttributes<'svg'> {
   name: IconName;
 };
 
-const { name, ...props } = Astro.props;
-const ariaLabel = props['aria-label'];
-const attributes = { 
-  ...props,
-  'aria-hidden': !ariaLabel,
-  role: props.role 
-    ? props.role
-    : ariaLabel ? 'img' : 'presentation',
-};
+const { 
+  name,
+  'aria-label': ariaLabel, 
+  ...props 
+} = Astro.props;
 ---
 
-<svg { ...attributes } data-icon={name}>
-  <use href={`#${name}`} xlink:href={`#${name}`} />
+<svg 
+  data-icon={name} 
+  role={ariaLabel ? 'img' : 'presentation'} 
+  aria-hidden={!ariaLabel} 
+  aria-label={ariaLabel} 
+  {...props}
+>
+  <use href={`#${name}`} />
```

Comment:
Are these not identical to the previous implementation besides removing `xlink:href`?